tasks/image-segmentation/src/train.py

```diff
@@ -0,0 +1,142 @@
+
+from src import detect
+from src.model import UNetModel
+from src.dataset import loadDataset, createBatches
+from src.callbacks import DisplayCallback
+
+
+def saveLiteModel(model: KerasModel, path: Path) -> None:
+    converter = tf.lite.TFLiteConverter.from_keras_model(model)
+    tfLiteModel = converter.convert()
+
+    with path.open("wb") as file:
+        file.write(tfLiteModel)
+
+
+def saveCoremlModel(model: KerasModel, path: Path) -> None:
+    model = coremltools.converters.convert(model)
+    model.save(str(path))
+
+
+def saveTFJSModel(modelPath: Path, path: Path) -> None:
+    tfjs.converters.convert_tf_saved_model(
+        str(modelPath),
+        str(path)
+    )
+
+
+def saveModelDescriptor(
+    model: KerasModel,
+    taskRun: TaskRun[ImageDataset],
+    coretexModel: Model,
+    path: Path
+) -> None:
+
+    labels = [
+        {
+            "label": clazz.label,
+            "color": clazz.color
+        }
+        for clazz in taskRun.dataset.classes
+    ]
+
+    coretexModel.saveModelDescriptor(path, {
+        "project_task": taskRun.projectType,
+        "labels": labels,
+        "modelName": coretexModel.name,
+        "description": taskRun.description,
+
+        "input_description": "RGB image",
+        "input_shape": model.input_shape,
+
+        "output_description": "Segmentation mask",
+        "output_shape": model.output_shape
+    })
+
+
+def train(taskRun: TaskRun) -> None:
+    taskRun.createMetrics([
+        Metric.create("loss", "epoch", MetricType.int, "value", MetricType.float, [0, taskRun.parameters["epochs"]]),
+        Metric.create("accuracy", "epoch", MetricType.int, "value", MetricType.float, [0, taskRun.parameters["epochs"]], [0, 1]),
+        Metric.create("val_loss", "epoch", MetricType.int, "value", MetricType.float, [0, taskRun.parameters["epochs"]]),
+        Metric.create("val_accuracy", "epoch", MetricType.int, "value", MetricType.float, [0, taskRun.parameters["epochs"]], [0, 1])
+    ])
+
+    taskRun.updateStatus(TaskRunStatus.inProgress, "Downloading dataset")
+    taskRun.dataset.download()
+
+    excludedClasses: list[str] = taskRun.parameters["excludedClasses"]
+    logging.info(f">> [Image Segmentation] Excluding classes: {excludedClasses}")
+    taskRun.dataset.classes.exclude(excludedClasses)
+
+    count, dataset = loadDataset(taskRun.dataset, taskRun)
+    trainCount, trainBatches, testCount, testBatches = createBatches(
+        dataset,
+        count,
+        taskRun.parameters["validationSplit"],
+        taskRun.parameters["batchSize"],
+        taskRun.parameters["imageSize"]
+    )
+
+    # + 1 because we also have a background class
+    classCount = len(taskRun.dataset.classes) + 1
+    model = UNetModel(classCount, taskRun.parameters["imageSize"])
+
+    taskRun.updateStatus(TaskRunStatus.inProgress, "Training the model")
+
+    epochs: int = taskRun.parameters["epochs"]
```

Comment:
As some parameters like `epochs` are no longer set to be required, this could lead to unexpected errors if the user leaves them at `null`. To prevent this, can you add a check for `None` for each of these parameters and throw a clear error message.